### DISCUSSION TITLE
how to check the implementation of particle dissolution method of PKSIM

### DISCUSSION THREAD

**Discussion by hainiwen** [Open-Systems-Pharmacology/Forum#1530] [Questions & Problems] [answered]
Hi there!

I want to use the particle dissolution function of PKSIM, however, I wonder how PKSIM implements it. 
I am aware of the publication is 
https://pubmed.ncbi.nlm.nih.gov/20554023/
But I still don't get how PKSIM compute it.

For example, does the number of bins equal to k that was talked about in the paper? meaning, how many groups of particles were assigned into?

Also, in the paper, it seems to be partial differential equations where both location and time are involved in, what are algorithms like in PKSIM? Should it be approximated to ODE to implement?

How and where can I find these equations so that I understand what's going on better?

Lots of thanks!!

Haini

**Comment by hainiwen** [comment]
In particular, I wonder how the method computes a Number Of Particles Factor to compute the number of particles based on the distribution of particles. 
I read from several references that it should relate to pdf distribution of radius of particles, but I am also quite lost there since its quite a strange area to me...
maybe the formula is somewhere in PKSIM so that I understand how come can be computed.

Cheers,
Haini

**Reply by Yuri05** [reply, marked as answer]
Number of particle bins is **user input**. Particles radius within each bin is estimated from the properties of the radius probability distribution, [min..max] radius range and the number of bins. 

E.g. if you have 3 bins, **normal** distribution, it would look like below
<img height="250" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/b1f0d472-7ccb-410a-8ffb-928b12fa4ad8"> <img height="250" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/29525a50-a983-4a49-b30f-bee88ad53916">

If you have 3 bins, **lognormal** distribution, it would look like below:
<img height="250" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/93379aaa-348e-40a8-908a-1cbb5a1f4a79"> <img height="250" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/c0f31dde-f43a-4c37-b41a-2524d8c8bf97">

Number of particles in the i-th particle bin ($N_i$ ) is assumed to be _proportional to the probability density function for the bin radius_ ($f(r_i)$ ):
$$N_i=alpha*f(r_i)$$
Knowing the total administered dose you can estimate the proportionality factor $alpha$ from the relationship
$$Dose = \sum_{i=1}^{Number\ of\ bins} 4/3\*\pi\*r_i^3\*\rho\*N_i$$
where $\rho$ is the density of drug particles.

From this you can estimate the number of particles in each bin as 
$$N_i = \frac{Dose}{\rho}\*\left[ \frac{f(r_i)}{4/3\*\pi\* \sum \left( r_i^3\*f(r_i) \right)} \right]$$

The expression in the square brackets above is defined as `Number_Of_Particles_Factor` and is calculated only in PK-Sim (because probability density functions are not supported by MoBi).

So finally the number of particles in each bin is defined as 
$$N_i = \frac{Dose}{\rho}\*Number\ Of\ Particles\ Factor = \frac{DrugMass * MolWeight}{\rho}\*Number\ Of\ Particles\ Factor$$
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/0818ae13-612c-403a-a872-563a03718203)

**Comment by Yuri05** [comment]
Regarding the overall implementation: the 2010 paper is outdated, current implementation is different.

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/4a20f586-f762-4803-a7fb-e302b35e1bd6)

Particles in each bin are moving along the GI-tract. The movent is realized by calculation of the **Number of particles fraction** in each GI segment. 

At t=0 **Number of particles fraction stomach** = 1 and **Number of particles fraction other segments** = 0
Then the particles start to distribute (**Number of particles fraction stomach** decreases, **Number of particles fraction duodenum** increases, etc.)

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/0a8ed61c-2505-47f4-b575-8d3406a15584)

The Noyes-Whitney dissolution is then realized **separately for each bin and each GI-segment**.

In the events building block you can see a dissolution transport **template** for each GI segment (Solid<=>Dissolved):
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/b010150d-2d80-4217-97ae-62dc6cf2dd8b)

When creating a simulation, each of those dissolution transports is created for each particles bin.

**Reply by hainiwen** [reply]
Thank you so much!!! That's very helpful! (and complicated )
